### PR TITLE
	internal/ethapi, internal/web3ext: adds raw tx retrieval methods

### PR DESCRIPTION
**Pull request by kobigurk**

### CODE REVIEW

**Comment by fjl** [comment]
Thanks. How is this useful? We already return all tx information as JSON fields so you could just recreate the transaction RLP on the client side...

**Comment by kobigurk** [comment]
The motivation was having a simple straightforward tool to do it instead of using an external one. Can also be useful if you want to transmit your tx with an external tool (such as etherscan's pushtx), but feel confident with the geth console.

**Comment by fjl** [comment]
The reason why I'm arguing against this feature is that it doubles the size of transaction JSON for all queries. I see how it can be useful, but maybe we should add a separate method to get the RLP version. It could be called `eth_getRawTransaction`.

**Comment by kobigurk** [comment]
I think it's a good idea - I felt the uneasiness too.

**Comment by fjl** [comment]
So, will you add `eth_getRawTransaction` instead?

**Comment by kobigurk** [comment]
Yeah, that's the plan!

**Comment by fjl** [comment]
Yes that sounds like the right thing.

**Comment by fjl** [comment]
Maybe we also need the other variants.

**Comment by kobigurk** [comment]
Good point. I will take a closer look at those too.

**Comment by kobigurk** [comment]
This adds support for the discussed methods. I intentionally didn't implement it in ethclient, as you already get the transaction object there, from which it's direct to get the raw bytes.

File: internal/jsre/ethereum_js.go [hunk 1]
```diff
@@ -5147,6 +5147,10 @@ var transactionFromBlockCall = function (args) {
     return (utils.isString(args[0]) && args[0].indexOf('0x') === 0) ? 'eth_getTransactionByBlockHashAndIndex' : 'eth_getTransactionByBlockNumberAndIndex';
 };
 
+var rawTransactionFromBlockCall = function (args) {
```
Comment (fjl) [inline internal/jsre/ethereum_js.go]: Please put web3.js extensions into [internal/web3ext/web3ext.go](https://github.com/ethereum/go-ethereum/blob/develop/internal/web3ext/web3ext.go)

Comment (kobigurk) [inline internal/jsre/ethereum_js.go]: done

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: c6ac7dae56e7d8483373847a830e226d06ed4f20
  * Commit subjects should be kept under 80 characters

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

File: internal/ethapi/api.go [hunk 2]
```diff
@@ -710,6 +720,18 @@ func newRPCTransaction(b *types.Block, txHash common.Hash) (*RPCTransaction, err
 	return nil, nil
 }
 
+// newRPCRawTransaction returns the bytes of a given transaction.
+func newRPCRawTransaction(tx *types.Transaction) (rpc.HexBytes, error) {
```
Comment (kobigurk) [inline internal/ethapi/api.go]: very true, done

**Pull request merged**